filesystem/index.go

```diff
@@ -0,0 +1,158 @@
+package fs
+
+import (
+	"context"
+	"io"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/ipfs/go-unixfs"
+)
+
+// This file represents a partial and theoretical
+// pkg-level implementation of a filesystem{}
+
+var (
+	pkgRoot FileSystem
+	closers []io.Closer
+)
+
+func init() {
+
+	// we depend on data from the coreapi to initalize our API nodes
+	// so fetch it or something and store it on the FS
+	daemon := fallbackApi()
+	ctx := deriveCtx(daemon.ctx) // if the daemon cancels, so should we
```

Comment:
When you'll be constructing fallback api, you'll control the context. https://github.com/ipfs/interface-go-ipfs-core/issues/32